Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp

```diff
@@ -514,6 +541,38 @@ namespace AZ
                 return true;
             }
 
+            void BootstrapSystemComponent::SwitchRenderPipeline(const AZ::RPI::RenderPipelineDescriptor& newRenderPipelineDesc, AZ::RPI::ViewportContextPtr viewportContext)
+            {
+                AZ::RPI::RenderPipelineDescriptor pipelineDescriptor = newRenderPipelineDesc;
+                pipelineDescriptor.m_name =
+                    AZStd::string::format("%s_%i", pipelineDescriptor.m_name.c_str(), viewportContext->GetId());
+
+                if (pipelineDescriptor.m_renderSettings.m_multisampleState.m_customPositionsCount &&
+                    !RHI::RHISystemInterface::Get()->GetDevice()->GetFeatures().m_customSamplePositions)
+                {
+                    // Disable custom sample positions because they are not supported
+                    AZ_Warning(
+                        "BootstrapSystemComponent",
+                        false,
+                        "Disabling custom sample positions for pipeline %s because they are not supported on this device",
+                        pipelineDescriptor.m_name.c_str());
+                    pipelineDescriptor.m_renderSettings.m_multisampleState.m_customPositions = {};
+                    pipelineDescriptor.m_renderSettings.m_multisampleState.m_customPositionsCount = 0;
+                }
+
+                // Create new render pipeline
+                auto oldRenderPipeline = viewportContext->GetRenderScene()->GetDefaultRenderPipeline();
+                RPI::RenderPipelinePtr newRenderPipeline = RPI::RenderPipeline::CreateRenderPipelineForWindow(
+                        pipelineDescriptor, *viewportContext->GetWindowContext().get(), AZ::RPI::ViewType::Default);
+
```

Comment:
Can you not just call LoadPipeline and then just add the code to remove the old pipeline?

Reply:
I think it's better to have the API function to accept RenderPipelineDescriptor than a product asset string. This way, user may use code to switch to a render pipeline without an actual asset file. Also, I think the switching render pipeline shouldn't need as strict as start up which requires the asset must exist. It can fail as long as we have feedback when it failed.

Reply:
I was mainly wondering about the duplicated code in this method and BootstrapSystemComponent::LoadPipeline and was wondering if we should consider refactoring it. But it's a minor point.